src/components/AgreementLineSections/Info.js

```diff
@@ -80,29 +83,45 @@ const Info = ({
         <Headline size="large" tag="h3">
           <FormattedMessage id="ui-agreements.eresources.packageDetails" />
         </Headline>
-        <PackageCard pkg={resource} />
+        { isExternal(line) ?
+          <PackageCardExternal pkg={resource} />
+          :
+          <PackageCard pkg={resource} />
+        }
       </>
       :
       <>
         <Headline size="large" tag="h3">
           <FormattedMessage id="ui-agreements.eresources.titleDetails" />
         </Headline>
         <KeyValue label={<FormattedMessage id="ui-agreements.eresources.titleOnPlatformURL" />}>
-          {resource.pti?.url ?
+          <div data-test-agreement-line-url>
+            { isExternal(line) ?
+          resource.reference_object?.url ?
+            <a href={resource.reference_object?.url} rel="noopener noreferrer" target="_blank">{resource.reference_object?.url}</a>
+            :
+            <NoValue />
+              :
+          resource.pti?.url ?
             <a href={resource.pti?.url} rel="noopener noreferrer" target="_blank">{resource.pti?.url}</a>
             :
             <NoValue />
           }
+          </div>
         </KeyValue>
-        <TitleCard title={resource} />
-        { isExternal(line) ? null : (
-          <>
-            <Headline size="large" tag="h3">
-              <FormattedMessage id="ui-agreements.eresources.parentPackageDetails" />
-            </Headline>
-            <PackageCard pkg={resource.pkg} />
-          </>
-        )}
+        { isExternal(line) ?
+          <TitleCardExternal title={resource} />
+          :
+          <TitleCard title={resource} />
+        }
+        <Headline size="large" tag="h3">
+          <FormattedMessage id="ui-agreements.eresources.parentPackageDetails" />
+        </Headline>
+        { isExternal(line) ?
+          <PackageCardExternal packageData={resource.reference_object.packageData} />
```

Comment:
Should we add in a `?`  check here for `resource.reference_object?.packageData` here? or is it assumed that `reference_object` always exists on the `resource` object?

Reply:
`reference_object`  should always exist but I added a `?` anyway